platforms/jvm/platform-jvm/src/main/java/org/gradle/jvm/tasks/Jar.java

```diff
@@ -64,6 +67,25 @@ public Jar() {
         metaInf = (CopySpecInternal) getRootSpec().addFirst().into("META-INF");
         metaInf.addChild().from(manifestFileTree());
         getMainSpec().appendCachingSafeCopyAction(new ExcludeManifestAction());
+        declareGracefulDegradationCondition();
+    }
+
+    private void declareGracefulDegradationCondition() {
+        if (!isExecutionTimeAccess()) {
```

Comment:
We need to ensure not to use `Project` to access the service registry (to obtain `ConfigurationCacheGracefulDegradationController`).